includes/message.message.inc

```diff
@@ -172,32 +172,35 @@ class Message extends Entity {
     if (!empty($arguments)) {
       $args = array();
       foreach ($arguments as $key => $value) {
-        if (is_array($value) && !empty($value['callback']) && function_exists($value['callback'])) {
-          // A replacement via callback function.
-          $value += array('pass message' => FALSE);
-          if ($value['pass message']) {
-            // Pass the message object as-well.
-            $value['callback arguments'][] = $this;
+        // Don't bother processing for keys not present in the output
+        if (strpos($output, $key) !== FALSE) {
```

Comment:
I guess this is the change, right? It's smart!

To keep the diff smaller - and I tend to prefer return early you can

```php
if (strpos($output, $key) === FALSE) {
  // The replacement key is not in the output, so save computation.
  continue;
}
```

I wonder if there are cases where we would still want to computation. We can add a `force_computation` (or another better name), in the `$options`.

(This will require a simpleTest)

Reply:
Thanks for the feedback; yes, the change is just a one-line check and comment wrapping the business inside the foreach. I like your approach, I'll incorporate that.

I also share your concern that argument callbacks might be (mis)used in a way that creates needed side effects. Adding an option to force callbacks to be executed anyway would allow any such misuers to update to force legacy behavior.